src/Control/Bind.purs

```diff
@@ -73,11 +73,11 @@ foreign import arrayBind :: forall a b. Array a -> (a -> Array b) -> Array b
 -- |
 -- | An example is the `Unit` type, since there is only one
 -- | possible value which can be returned.
-class Discard a where
-  discard :: forall f b. Bind f => f a -> (a -> f b) -> f b
+class BindDiscard a where
+  bind_ :: forall f b. Bind f => f a -> (a -> f b) -> f b
```

Comment:
No, it can't have that type, the `b` is the final result of the `do` block. I was just saying that `bind_` suggests that to me, so it might be surprising.